OpenRA.Game/GameRules/WeaponInfo.cs

```diff
@@ -221,20 +221,23 @@ public bool IsValidAgainst(FrozenActor victim, Actor firedBy)
 		}
 
 		/// <summary>Applies all the weapon's warheads to the target.</summary>
-		public void Impact(Target target, WarheadArgs args)
+		public void Impact(in Target target, WarheadArgs args)
 		{
 			var world = args.SourceActor.World;
 			foreach (var warhead in Warheads)
 			{
 				if (warhead.Delay > 0)
-					world.AddFrameEndTask(w => w.Add(new DelayedImpact(warhead.Delay, warhead, target, args)));
+				{
+					var target1 = target;
```

Comment:
Can we have a comment here like `// Lambdas can't use in variables, so capture a copy for later` and rename this to something like `delayedTarget`?